### PR TITLE
fix: mcp agent policies and deduplicate blocks when merging unrolled assistants

### PR DESCRIPTION
**Pull request by RomneyDa**
## Description
Currently agents fail because default account models have a secrets issue
    

---

## Summary by cubic
Deduplicates blocks when merging unrolled assistants to prevent duplicate models, rules, prompts, and more. Incoming blocks take priority, and env/requestOptions are merged predictably.

- **Bug Fixes**
  - Merge arrays across all block types (models, context, rules, prompts, docs, data, mcpServers) with deduplication via BlockDuplicationDetector; incoming items are kept, duplicates are filtered.
  - Filter nulls and set block arrays to undefined when empty after merge.
  - Merge env with incoming overriding base; merge requestOptions with base preferred and headers combined.
  - Add comprehensive Jest tests covering dedup, ordering, env/requestOptions behavior, and edge cases.

### CODE REVIEW

File: extensions/cli/src/services/ToolPermissionService.ts [hunk 1]
```diff
@@ -61,55 +64,80 @@ export class ToolPermissionService
...
+          tool: t.name,
+          permission: "exclude",
+        }));
+        policies.push(...disallowed);
       }
     }
 
-    if (policies.length > 0) {
-      logger.debug(`Generated ${policies.length} agent file tool policies`);
+    const hasMcp = !!parsedTools.mcpServers?.length;
```
Comment (sestinj) [inline extensions/cli/src/services/ToolPermissionService.ts:130]: i feel like this is pretty dense code. if it has to be that dense, could it get an extra comment or two?

**Review by sestinj — CHANGES_REQUESTED** [review]

**Review by sestinj — APPROVED** [review]
